modules/features/discover/src/main/java/au/com/shiftyjelly/pocketcasts/discover/viewmodel/DiscoverViewModel.kt

```diff
@@ -111,6 +112,31 @@ class DiscoverViewModel @Inject constructor(
             }
     }
 
+    fun loadCarouselSponsoredPodcasts(
+        sponsoredPodcastList: List<SponsoredPodcast>
+    ): Flowable<List<CarouselSponsoredPodcast>> {
+        val sponsoredPodcastsSources = sponsoredPodcastList
+            .filter { it.source != null && it.position != null }
```

Comment:
If this is something that should never happen, would it be worth logging an error here if it ever does happen?

Reply:
Logged it in 93f0900